tests/integration/techRecords.intTest.ts

```diff
@@ -290,6 +293,237 @@ describe("techRecords", () => {
+                newTechRec.vin = Date.now().toString();
+                newTechRec.partialVin = newTechRec.vin.substr(newTechRec.vin.length - 6);
+                newTechRec.techRecord[0].bodyType.description = "New Tech Record";
+                newTechRec.primaryVrm = Math.floor(100000 + Math.random() * 900000).toString();
+                request.post("vehicles")
+                  .send(newTechRec)
+                  .end((err, res: any) => {
+                    expect(res.statusCode).toEqual(201);
+                    expect(res.headers["access-control-allow-origin"]).toEqual("*");
+                    expect(res.headers["access-control-allow-credentials"]).toEqual("true");
+                    expect(res.body).toEqual("Technical Record created");
+                    done();
+                  });
+              });
+            });
+
+            context("and that vehicle does exist", () => {
+              it("should return status 400 The conditional request failed", (done) => {
+                request.post("vehicles")
+                  .send(newTechRec)
+                  .end((err, res: any) => {
+                    expect(res.statusCode).toEqual(400);
+                    expect(res.headers["access-control-allow-origin"]).toEqual("*");
+                    expect(res.headers["access-control-allow-credentials"]).toEqual("true");
+                    expect(res.body).toEqual("The conditional request failed");
+                    done();
+                  });
+              });
+            });
+          });
+
+          context("and the payload is invalid", () => {
+            context("and the techRecord array is empty", () => {
+              it("should return status 400 The conditional request failed", (done) => {
+                newTechRec.techRecord = [];
+                request.post("vehicles")
+                  .send(newTechRec)
+                  .end((err, res: any) => {
+                    expect(res.statusCode).toEqual(400);
+                    expect(res.headers["access-control-allow-origin"]).toEqual("*");
+                    expect(res.headers["access-control-allow-credentials"]).toEqual("true");
+                    expect(res.body).toEqual("Body is not a valid TechRecord");
+                    done();
+                  });
+              });
+            });
+          });
+        });
+      });
+    });
+  });
+
+  describe("updateTechRecords", () => {
+    context("when database is populated", () => {
+      beforeAll(async () => {
+        await emptyDatabase();
+      });
+
+      afterAll(async () => {
+        await populateDatabase();
+      });
 
+      context("when database is populated", () => {
+        const convertToResponse = (dbObj: any) => { // Needed to convert an object from the database to a response object
+          const responseObj = Object.assign({}, dbObj);
+
+          // Adding primary and secondary VRMs in the same array
+          const vrms: any = [{isPrimary: true}];
```

Comment:
related to question above - this looks like it's fixing a data issue which is better fixed at the source